### ISSUE TITLE
Background not well removed

### ISSUE THREAD

**Issue by Jordi100** [OPHoperHPO/image-background-remove-tool#9]
Hello,

First, thank you for share this code. I appreciate.
I have installed it using python 3.7 and tensorflow 2.1
But when I try to remove the background from leaves, sometimes the background is not removed or removed partially. Also, sometimes part of the leaf is removed. So it seems it still needs some accuracy, some precision.
Please, can you tell me which parameter should I modify to make the background could be detected better? I have checked with 16 images of leaves and only one image have had the background removed as expected.
Regards,

Jorge

**Comment by OPHoperHPO** [comment]
Hello, @Jordi100.
It looks like you are using an old version of the program.  
Try updating it. Today a new version with a new neural network was released.
The old version of the program and the Tensorflow models in the new version are designed solely to remove the background from portrait photographs (photographs with people) and photographs with animals. 
Try using the 'u2net' model for your purposes.  If the result is still bad or even worse, try to train your model solely on what you need to remove the background from, this in theory should significantly improve the result.

**Comment by Munawwar** [comment]
Here is my approach so far to improve the background removal:

```python
import cv2
...
...

image = model.process_image("u2net")
mask = np.array(image.getchannel('A'))

# create a better mask
mask[mask < 128] = 0
mask[mask >= 128] = 255

src = cv2.imread(inputpath, 1)

# ------- do an alpha blended cutout ---------
# first by blurring the borders
# TODO: make erode/blur scale with image scale?
mask2 = cv2.erode(mask, np.ones((15,15),np.uint8) , iterations = 1)
mask3 = cv2.GaussianBlur(mask2, (15,15), 0)
alpha = mask3.astype(float) * 1.2  # making blend stronger
alpha[alpha > 255] = 255.0
# Normalize the alpha mask to keep intensity between 0 and 1
alpha = alpha / 255.0

# now.. cutting time!
foreground = np.copy(src).astype(float)
# Multiply the foreground with the alpha matte
foreground = cv2.multiply(alpha, foreground)

background = np.ones_like(foreground, dtype=float) * 255.0
# Multiply the background with ( 1 - alpha )
background = cv2.multiply(1.0 - alpha, background)

# Add the masked foreground and background.
finalImage = cv2.add(foreground, background)
finalImage[finalImage > 255] = 255
finalImage = finalImage.astype(np.uint8)
```

Shall I add this as a CLI option?

**Comment by Jordi100** [comment]
Hello,

Thank you for your answer. Sorry for my late reply.
Yes, with the new version it has improved a lot. Most of the backgrounds are removed.
I will pay you a coffee with the platform you use.
Thank you.

Jorge
